docs/metastore_upgrade_guide.md

```diff
@@ -0,0 +1,224 @@
+To effectively upgrade the metastores, four principal operations are required:
+1. Assess - In this step, you evaluate the existing HMS tables identified for upgrade to determine the right approach for upgrade. This step is a prerequisite and is performed by the assessment workflow.
+2. Create - In this step, you create the required UC assets such as, Metastore, Catalog, Schema, Storage Credentials, External Locations. This step is part of the upgrade process.
+3. Upgrade/Grant these are two steps that UCX combine.
+   4. Upgrade - The metastores objects (tables/views) will be converted to a format supported by UC 
+   4. Grant - The table upgrade the newly created object the same permission as the original object.
+
+## Prerequisites
+For UCX to be able to upgrade the metastore. The following prerequisites must be met:
+1. UCX must be installed and configured on the workspace. For more information on how to install UCX, refer to the [external_hms_glue.md](external_hms_glue.md).
+2. In case of an external metastore (such as GLUE), UCX has to be configured to attach to the metastore. For more information on how to configure UCX to attach to an external metastore, refer to the [External Metastore Guide]().
+3. The assessment workflow must be run.
+4. It is recommended that the group migration process will be completed before upgrading the metastore. For more information on how to migrate groups, refer to the [UCX Readme](../README.md).
+5. The workspace should be configured with a Metastore follow the instruction here [Create UC Metastore](https://docs.databricks.com/en/data-governance/unity-catalog/create-metastore.html)<br>
+   Metastore can be attached to the workspace using the following UCX command:<br>
+   ```bash
+    databricks labs ucx assign-metastore --workspace-id <workspace-id> [--metastore-id <metastore-id>]
+    ```
+
+## Upgrade Process
+The upgrade process is done in multiple steps. For each step we will discuss the manual process and how to perform it using UCX.
+
+### Step 1: Mapping Metastore Tables (UCX Only)
+In this step we will map the metastore tables to UC tables.
+#### Step 1.1: Create the mapping file
+This step can be performed using the `create-table-mapping` command documented in the [UCX Readme](../README.md#create-table-mapping-command).
+CLI Command
+```bash
+databricks labs ucx create-table-mapping 
+```
+
+The CLI command will create a mapping file in the workspace UCX folder.
+
+The format of the mapping file is as follows:
+
+| **columns:** | **workspace_name**      | **catalog_name** | **src_schema** | **dst_schema** | **src_table** | **dst_table** |
+|--------------|---------------------|--------------|----------------|----------------|---------------|---------------|
+| values:      | data_engineering_ws | de_catalog   | database1      | database1      | table1        | table1        |
+
+By default, all the tables/views will be mapped to UC tables.
```

Comment:
can you rather expand the CLI command sections?

Reply:
Done